ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/SessionStoreCacheBypass.java

```diff
@@ -143,6 +152,85 @@ private SessionStoreCacheBypass() {}
     }
   }
 
+  @SuppressWarnings("unchecked")
+  public static KeyValueIterator<Windowed<GenericKey>, GenericRow> fetchRange(
+          final ReadOnlySessionStore<GenericKey, GenericRow> store,
+          final GenericKey keyFrom,
+          final GenericKey keyTo
+  ) {
+    Objects.requireNonNull(keyFrom, "lower key can't be null");
+    Objects.requireNonNull(keyTo, "upper key can't be null");
+
+    final StateStoreProvider provider;
+    final String storeName;
+    final QueryableStoreType<ReadOnlySessionStore<GenericKey, GenericRow>> storeType;
+    try {
```

Comment:
Can you not pull out some of this logic into a `getStores` as you did in the windowstorecachebypass?

Reply:
it can be. it is done